cloud-controller-manager/osc/instances_v2.go

```diff
@@ -255,6 +264,11 @@ func parseInstanceIDFromProviderIDV2(providerID string) (string, error) {
 	} else if len(metadata) == 3 {
 		// /az/instance-id
 		instanceID = metadata[2]
+	} else {
+		return "", errors.New("invalid providerID format")
+	}
+	if instanceID == "" {
+		return "", fmt.Errorf("instance ID not found in providerID: %s", providerID)
```

Comment:
Unreachable code with the `else`, no ?

Reply:
what if  providerId "aws://":
strings.TrimPrefix(providerID, "aws://") results in "" 
=> len(metadata) == 1 => instanceID = metadata[0]="" => so instanceId will not be found 
Am I right ?

Reply:
You're right, the providerId needs to be check in all cases